configure.ac

```diff
@@ -6157,6 +6157,10 @@ PY_STDLIB_MOD([fcntl],
   [], [$FCNTL_LIBS])
 PY_STDLIB_MOD([mmap],
   [], [test "$ac_cv_header_sys_mman_h" = "yes" -a "$ac_cv_header_sys_stat_h" = "yes"])
+PY_STDLIB_MOD([_socket],
+  [], m4_flatten([test "$ac_cv_header_sys_socket_h" = "yes"
+                    -a "$ac_cv_header_sys_types_h" = "yes"
+                    -a "$ac_cv_header_netinet_in_h" = "yes"]))
```

Comment:
While I appreciate your thoroughness, I don't think this check is necessary. We have been building socket by default for years and assume that basic socket features are always available.